x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/lookup/QueryList.java

```diff
@@ -63,42 +59,94 @@ int getPositionCount() {
     /**
      * Returns a list of term queries for the given field and the input block.
      */
-    static QueryList termQueryList(
-        MappedFieldType field,
-        SearchExecutionContext searchExecutionContext,
-        Block block,
-        DataType inputDataType
-    ) {
-        return new TermQueryList(field, searchExecutionContext, block, inputDataType);
+    public static QueryList termQueryList(MappedFieldType field, SearchExecutionContext searchExecutionContext, Block block) {
```

Comment:
After the changes here, this function name/javadoc may be misleading. It's no longer a "generic" entrypoint for term queries.

I would maybe rename it to something like... `rawTermQueryList()`? Or something else. And add some javadoc explaining that for specific types like geo and datetime, the other functions should be used

Reply:
:+1: